src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -1133,14 +1133,29 @@ void QgsLayoutItemLegend::doUpdateFilterByMap()
 
     mapSettings.setExpressionContext( createExpressionContext() );
 
-    const QgsGeometry atlasGeometry = mInAtlas ? mLayout->reportContext().currentGeometry( mapSettings.destinationCrs() ) : QgsGeometry();
+    const QgsGeometry atlasGeometry { mLayout->reportContext().currentGeometry( mapSettings.destinationCrs() ) };
```

Comment:
if we remove the condition on `mInAtlas`, on clipping with coming from `atlasClippingSettings()` would work, not the one coming from the `mFilterLegendByAtlasCheckBox` in `QgsLayoutLegendWidget` ([see](https://github.com/qgis/QGIS/blob/master/src/gui/layout/qgslayoutlegendwidget.cpp#L1226) ), no?

Reply:
I didn't really remove the condition, I just moved down to where it was used: https://github.com/qgis/QGIS/pull/54692/files#diff-e7d9625a78a678e8657e01df0b53ef2c124469d2a30ef3977c13a8bc03ce14f1R1178  

IMHO it is much clearer than relying on the null geometry for the `if` banching.